Makefile

```diff
@@ -11,40 +11,53 @@ test: test-containerized st                             ## Run all the tests
 ###############################################################################
 
 ###############################################################################
-# The build architecture is select by setting the ARCH variable.
-# For example: When building on ppc64le you could use ARCH=ppc64le make <....>.
-# When ARCH is undefined it defaults to amd64.
-ARCH?=amd64
-ifeq ($(ARCH),amd64)
-	ARCHTAG?=
-	GO_BUILD_VER:=v0.12
+# Both native and cross architecture builds are supported.
+# The target architecture is select by setting the ARCH variable.
+# When ARCH is undefined it is set to the detected host architecture.
+# When ARCH differs from the host architecture a crossbuild will be performed.
+ARCHES=$(patsubst Dockerfile.%,%,$(wildcard Dockerfile.*))
+
+# BUILDARCH is the host architecture
+# ARCH is the target architecture
+# we need to keep track of them separately
+BUILDARCH ?= $(shell uname -m)
+BUILDOS ?= $(shell uname -s | tr A-Z a-z)
+
+# canonicalized names for host architecture
+ifeq ($(BUILDARCH),aarch64)
+        BUILDARCH=arm64
 endif
-
-ifeq ($(ARCH),ppc64le)
-	ARCHTAG:=-ppc64le
-	GO_BUILD_VER:=latest
+ifeq ($(BUILDARCH),x86_64)
+        BUILDARCH=amd64
 endif
 
-ifeq ($(ARCH),s390x)
-	ARCHTAG:=-s390x
-	GO_BUILD_VER:=latest
+# unless otherwise set, I am building for my own architecture, i.e. not cross-compiling
+ARCH ?= $(BUILDARCH)
+
+# canonicalized names for target architecture
+ifeq ($(ARCH),aarch64)
+        override ARCH=arm64
+endif
+ifeq ($(ARCH),x86_64)
+    override ARCH=amd64
 endif
 
 # Determine which OS.
 OS := $(shell uname -s | tr A-Z a-z)
 
 ###############################################################################
+GO_BUILD_VER ?= v0.10
```

Comment:
Is there a reason we can't use v0.12 like we were before?

Reply:
I cannot think of a single good one, other than, to quote (was it you? cannot recall): "lazy copy-paste", but this time, on my part. :-)

Updating now.

Reply:
I had to revert back to v0.10; cross builds were failing on arm64 because of the `aarch64`/`arm64` issue in v0.12. See [this PR](https://github.com/projectcalico/go-build/pull/41)

Reply:
Updated to 0.14